ambry-network/src/main/java/com.github.ambry.network/Selector.java

```diff
@@ -548,3 +568,16 @@ private SocketChannel channel(SelectionKey key) {
     return (SocketChannel) key.channel();
   }
 }
+
+/**
+ * Class used to store {@link SSLTransmission} which are yet to complete its handshake, along with start time of the same
+ */
+class TransmissionPendingHandshake {
```

Comment:
As mentioned above, I think this can just be `UnreadyConnections`, and the member variable could be `Transmission` instead of `SSLTransmission`.